### PR TITLE
fnlfmt: init

### PR DESCRIPTION
**Pull request by gpanders**
###### Motivation for this change

[Fennel](https://fennel-lang.org) is now in nixpkgs, so it's useful to also have the canonical Fennel formatter.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

This is my first package for Nix, so please give me feedback and let me know if I could have done anything better.

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132612` at 75596c29 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/55aefaf07b045ad5b438c19784dee70a">fnlfmt</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by gpanders** [comment]
The `fnlfmt` Makefile uses a [vendored version of `fennel`](https://git.sr.ht/~technomancy/fnlfmt/tree/main/item/Makefile#L3) which won't work because that version of fennel has a hardcoded `#!/usr/bin/env` shebang. I just rewrote the `buildPhase` to do what the Makefile does but using the Nix provided version of `fennel` instead.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4001,6 +4001,16 @@
       fingerprint = "5214 2D39 A7CE F8FA 872B  CA7F DE62 E1E2 A614 5556";
     }];
   };
+  gpanders = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:4004]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/development/tools/fnlfmt/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    ${fennel}/bin/fennel --require-as-include --compile cli.fnl >> fnlfmt
+    chmod +x fnlfmt
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    patchShebangs ./fnlfmt
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fnlfmt/default.nix:31]: Why do we need to patch the shebang?

Comment (gpanders) [reply pkgs/development/tools/fnlfmt/default.nix:31]: Ah you're right, this is no longer necessary since we're not using the Makefile.
